lms/djangoapps/course_blocks/transformers/start_date.py

```diff
@@ -43,6 +43,26 @@ def name(cls):
         """
         return "start_date"
 
+    @classmethod
+    def _check_has_scheduled_content(cls, block_structure, scheduled_content_condition):
+        '''
+        Returns a block structure where the root course block has been
+        updated to include a has_scheduled_content field (True if the course
+        has any blocks with release dates in the future, False otherwise).
+        '''
+        has_scheduled_content = False
+        for block_key in block_structure:
```

Comment:
I would recommend looking into the different ways to sort through the block_structure. I believe there is a version that starts at the leaves (topological sort or something like that) and it might have a faster return time than this one that starts at the root and iterates down.